### PR TITLE
Standardize use of Logger

### PR DESCRIPTION
**Pull request by eric-maynard**
# Description

I noticed there was about 50/50 usage of `Logger LOG` and `Logger LOGGER` in the code, so I standardized on the more popular `Logger LOGGER`.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactor

# How Has This Been Tested?

_This PR does not change any behavior_

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snazy** [comment]
@eric-maynard can you rebase?
The change LGTM (and I personally prefer `LOGGER` ;) )

File: polaris-core/src/main/java/io/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public abstract class LocalPolarisMetaStoreManagerFactory<StoreType>
 
   protected PolarisStorageIntegrationProvider storageIntegration;
 
-  private Logger logger =
+  private Logger LOGGER =
```
Comment (adutra) [inline polaris-core/src/main/java/io/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java:54]:
```suggestion
  private static final Logger LOGGER =
```

Comment (ebyhr) [reply polaris-core/src/main/java/io/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java:54]: The following classes looks missing `static`:
* AzureCredentialStorageIntegrationTest 
* ManifestFileCleanupTaskHandler
* TracingFilter

Also, we should make the constant to `private` in the following classes:
* DefaultOAuth2ApiService
* PolarisApplicationIntegrationTest

**Review by ebyhr — APPROVED** [review]

**Comment by flyrain** [comment]
Are we considering third-party lib like Lombok? It will reduce code further. Here is an example, and we don't have to include spring to use Lombok log annotation. 
```
import lombok.extern.java.Log;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class LogExampleOther {
  
  public static void main(String... args) {
    log.error("Something else is wrong here");
  }
}
```

**Comment by eric-maynard** [comment]
@flyrain Right now my goal is just to rationalize the existing logging code without introducing a new library.

**Comment by snazy** [comment]
> Are we considering third-party lib like Lombok?

No Lombok, please. The source never matches the actual code, which makes debugging extremely hard.

**Comment by adutra** [comment]
> > Are we considering third-party lib like Lombok?
> 
> No Lombok, please. The source never matches the actual code, which makes debugging extremely hard.

Agreed, no Lombok please. For things like getters and setters, or generating equals() and hashCode(), there are better alternatives like [Immutables](https://immutables.github.io/immutable.html).

**Comment by flyrain** [comment]
I'm OK with any alternatives if they are better. Considering the immutability and generating a constructor, equals(), hashCode(), and toString() methods, Java Record(JDK16+) is a good one so that no third-party libs are needed.

**Review by collado-mike — APPROVED** [review]

**Comment by collado-mike** [comment]
I don't personally hate Lombok, but that discussion is out of scope of this PR, IMO

**Comment by adutra** [comment]
> I'm OK with any alternatives if they are better. Considering the immutability and generating a constructor, equals(), hashCode(), and toString() methods, Java Record(JDK16+) is a good one so that no third-party libs are needed.

True, but polaris-core is built for Java 11 currently, so records wouldn't be an option there. But as @collado-mike said, let's leave that discussion for later.

**Review by aihuaxu — APPROVED** [review]

**Review by dimas-b — APPROVED** [review]
LGTM :+1:

**Review by dimas-b — DISMISSED** [review]

**Review by dimas-b — DISMISSED** [review]

**Review by dimas-b — APPROVED** [review]

**Pull request merged**